src/Client.php

```diff
@@ -301,6 +323,28 @@ protected function getContext(ConnectionInterface $connection)
     }
 
     /**
+     * Derives a set of socket context options for a given secure connection.
+     *
+     * @param \Phergie\Irc\ConnectionInterface $connection
+     * @return array Associative array of context option key-value pairs
+     */
+    protected function getSecureContext(ConnectionInterface $connection)
+    {
+        $context = array();
+        if ($this->getForceIpv4Flag($connection)) {
+            $context['bindto'] = '0.0.0.0:0';
+        }
+        if ($this->getVerifyPeerFlag($connection)) {
+            $context['verify_peer'] = false;
+        }
+        if ($this->getAllowSelfSignedFlag($connection)) {
+            $context['allow_self_signed'] = true;
+        }
+        //$context = array('socket' => $context);
```

Comment:
At first glance, it looks like it would be possible to do so, but I think that may be better addressed in a separate PR once this one is merged.

Alternatively, we could have a PR that does nothing but update `react/socket-client` to `^0.5`, and this and the other prospective PR could be based against that one, so that the two can be developed concurrently.